Src/Newtonsoft.Json/JsonConverter.cs

```diff
@@ -96,4 +97,75 @@ public virtual bool CanWrite
             get { return true; }
         }
     }
+
+    /// <summary>
+    /// Converts an object to and from JSON.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public abstract class JsonConverter<T> : JsonConverter
+    {
+        /// <summary>
+        /// Writes the JSON representation of the object.
+        /// </summary>
+        /// <param name="writer">The <see cref="JsonWriter"/> to write to.</param>
+        /// <param name="value">The value.</param>
+        /// <param name="serializer">The calling serializer.</param>
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
```

Comment:
Yes, it is a breaking change for existing inheritors. But if you decide to take this PR, I'd strongly suggest to take it (next version will be a major version bump I think?). Otherwise the `override` semantics will be pretty unclear (e.g. overriding the generic method will have no effect if the non-generic method is also overridden).